### ISSUE TITLE
dvc monterey (intel) bottle contains bash completion script with extraneous text at the end, causing error on sourcing

### ISSUE THREAD

**Issue by aristocrates** [Homebrew/homebrew-core#122197]
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
$ brew config
HOMEBREW_VERSION: 3.6.20
ORIGIN: https://github.com/Homebrew/brew
HEAD: 344d32bf7f40183359592ef81ca6486e0587981b
Last commit: 2 weeks ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: f9ecf308945d7ec25a17c1c3ca3c8a4eb8a9890c
Core tap last commit: 40 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.cYYsM5qkma/org.macosforge.xquartz:0
HOMEBREW_EDITOR: vim
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit kabylake
Clang: 14.0.0 build 1400
Git: 2.39.1 => /usr/local/bin/git
Curl: 7.86.0 => /usr/bin/curl
macOS: 13.2-x86_64
CLT: 14.2.0.0.1.1668646533
Xcode: 14.2
```

```shell
$ brew doctor
Your system is ready to brew.
```

### Verification

- [X] My "`brew doctor` output" says `Your system is ready to brew.` and am still able to reproduce my issue.
- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.

### What were you trying to do (and why)?

After updating package `dvc`, I noticed that new bash sessions contained the following output:

```
-bash: $'\E[33mHaving': command not found
```

I traced the root cause to an issue with the bash completion script installed at `/usr/local/Cellar/dvc/2.43.2/etc/bash_completion.d/dvc`

### What happened (include all command output)?

The bottle for intel ventura with hash `a0ce3ade4df1f765e69ab8acc1c421a141dfbeba93aab63b4d7ceecfeb8b3343` (see https://github.com/Homebrew/homebrew-core/commit/eeec3f7e73a847a8aee0916c8a25bd81974ac3f1) seems to have a broken bash completion script due to the erroneous inclusion of a string from this upstream code-path https://github.com/iterative/dvc/blob/18fa1a0272826143f27db6a3b923eea149a519c5/dvc/logger.py#L10-L18

```
[...]

complete -o filenames -F _shtab_dvc dvc

ESC[33mHaving any troubles?ESC[39m Hit us up at ESC[34mhttps://dvc.org/support
ESC[39m, we are always happy to help!
```

If bash completion is sourced for dvc this results in the following warning from bash upon shell start:

```
-bash: $'\E[33mHaving': command not found
```

This seems to be the footer from this file https://github.com/iterative/dvc/blob/18fa1a0272826143f27db6a3b923eea149a519c5/dvc/logger.py#L10-L18

I am not sure how to actually make it so that `dvc completion` outputs that string and I am having trouble reproducing the file contents I see in the bottle by running the code path run by `generate_completions_from_executable` in a minimal reproducible example, so I'm not sure if there is a deeper root cause in https://github.com/Homebrew/brew or in https://github.com/iterative/dvc or if there is just some hiccup that happened when generating the bottle tarball.

https://github.com/Homebrew/brew/blob/3af2f0ececc848bca34ddbd5b876d20dc24a6aaa/Library/Homebrew/formula.rb#L1751-L1754

Other bottles may have the same issue but I have only confirmed this for the intel ventura bottle.

### What did you expect to happen?

Expected behavior: the bash completion script should be equal to the output of `dvc completion -s bash` and not contain any appended text at the end.

### Step-by-step reproduction instructions (by running `brew` commands)

Either:
* `brew install dvc`
* Inspect file contents of `/usr/local/Cellar/dvc/2.43.2/etc/bash_completion.d/dvc`

Or: (to just inspect the artifact that has the issue)
* `brew install oras`
* `oras pull ghcr.io/homebrew/core/dvc:2.43.2@sha256:47163ccc1b98ee4b52eecaca96df99c6cd6fa9d8b3972bf9075c2191487541c4` (per https://github.com/Homebrew/homebrew-core/pkgs/container/core%2Fdvc/67799726?tag=2.43.2)
    * In current working dir, file `dvc--2.43.2.ventura.bottle.tar.gz` should now be present
* `sha256sum dvc--2.43.2.ventura.bottle.tar.gz` (confirm this equals `a0ce3ade4df1f765e69ab8acc1c421a141dfbeba93aab63b4d7ceecfeb8b3343` which is the value for `:ventura` in https://github.com/Homebrew/homebrew-core/commit/eeec3f7e73a847a8aee0916c8a25bd81974ac3f1)
* `tar xf dvc--2.43.2.ventura.bottle.tar.gz`
* Inspect contents of `dvc/2.43.2/etc/bash_completion.d/dvc`

For me, I get the following checksum for that file from this bottle:

```
$ sha256sum dvc/2.43.2/etc/bash_completion.d/dvc
f006533e9d98c2e7b42e4e5f9f159b06508c6285797c178943ae95f9a0469b74  dvc/2.43.2/etc/bash_completion.d/dvc
```

`tail dvc/2.43.2/etc/bash_completion.d/dvc` gives:

```
    unset IFS
    COMPREPLY+=( $(compgen -W "${current_action_choices[*]}" -- "${completing_word}") )
  fi

  return 0
}

complete -o filenames -F _shtab_dvc dvc

ESC[33mHaving any troubles?ESC[39m Hit us up at ESC[34mhttps://dvc.org/support
ESC[39m, we are always happy to help!
```

This is the diff between the file included in the bottle and `dvc completion -s bash`

```diff
--- ./b/correct_bash_completion	2023-02-02 19:15:43
+++ ./dvc/2.43.2/etc/bash_completion.d/dvc	2023-02-02 03:18:59
@@ -2328,3 +2328,5 @@
 }
 
 complete -o filenames -F _shtab_dvc dvc
+
+[33mHaving any troubles?[39m Hit us up at [34mhttps://dvc.org/support[39m, we are always happy to help!
```

**Comment by aristocrates** [comment]
Note that if I run `brew reinstall --build-from-source dvc` the bash completion script is generated correctly

```sh
$ sha256sum /usr/local/Cellar/dvc/2.43.2/etc/bash_completion.d/dvc
23126f361b7fa6df2e94e89071b7792adbeed5a3071dc31ffa153a1f6ecdac1a  /usr/local/Cellar/dvc/2.43.2/etc/bash_completion.d/dvc
```

```
$ tail /usr/local/Cellar/dvc/2.43.2/etc/bash_completion.d/dvc
    COMPREPLY=( $([ -n "${current_action_compgen}" ] \
                  && "${current_action_compgen}" "${completing_word}") )
    unset IFS
    COMPREPLY+=( $(compgen -W "${current_action_choices[*]}" -- "${completing_word}") )
  fi

  return 0
}

complete -o filenames -F _shtab_dvc dvc
```

**Comment by gromgit** [comment]
It's actually worse than that: All macOS bottles are similarly affected, only the Linux bottle is "clean".

**Comment by aristocrates** [comment]
Looks like this is fixed (or at least mitigated) as of dvc 2.45.1 https://github.com/iterative/dvc/commit/ddeb32ed33afc15a4fb8f8b762bd7fe2cf07968d

See: https://github.com/iterative/dvc/issues/8956
